### ISSUE TITLE
Lombok in enums used as static imported constants : import order matters

### ISSUE THREAD

**Issue by maximebochon** [rzwitserloot/lombok#903]
I wrote this sample code to illustrate a quite annoying phenomon during compilation:

```java
package some.where;

import lombok.AllArgsConstructor;
import lombok.Getter;

import static some.where.UsingLombokWithEnums.TimeDirection.*;

import org.joda.time.LocalDate;

public class UsingLombokWithEnums {

	@AllArgsConstructor
	@Getter
	public enum TimeDirection {

		PAST(-1),
		FUTURE(+1);
		
		private final int value;
	}

	public static LocalDate move(LocalDate date, int dayCount, TimeDirection timeDirection) {
		return date.plusDays(timeDirection.getValue() * dayCount);
	}
	
	public static LocalDate moveForward(LocalDate date, int dayCount) {
		return move(date, dayCount, FUTURE);
	}
	
	public static LocalDate moveBackward(LocalDate date, int dayCount) {
		return move(date, dayCount, PAST);
	}
}
```

What is important here is that:
 - *enum* `TimeDirection` uses Lombok annotations
 - `PAST` and `FUTURE` *enum* constants are statically imported
 - these constants are used in static methods of the same class

Now the point is that **the import order matters**:
 - if Lombok imports appear before static imports, compilation will succeed
 - if Lombok imports appear after static imports, compilation will fail

(error message: *cannot find symbol* for `Getter` and `AllArgsConstructor`)

I'm aware that this kind of code plays with limits of Java compilation phase, in so far as Lombok itself is kind of a player. But I would like to get your opinion:
 - should this kind of code be avoided?
 - is it already documented somewhere? or a duplicate?
 - or is it a legitimate case that would need to be taken in account by next release?

From a practical point of view, in Eclipse (Luna), the *Organize Imports* functionality will place all static imports first, leading to compilation error (I didn't try in Netbeans or IntelliJ).

**Comment by davidghiurco** [comment]
Has this changed in the past 3 years? lol

**Comment by tej451** [comment]
Yes, I found this error today after refactoring (reorganising imports) of my project old code .

**Comment by Rawi01** [comment]
I cannot reproduce this anymore. @LiamClark Can you share an example?
